packages/clarity-js/src/data/upload.ts

```diff
@@ -39,35 +39,20 @@ export function queue(tokens: Token[], transmit: boolean = true): void {
         let now = time();
         let type = tokens.length > 1 ? tokens[1] : null;
         let event = JSON.stringify(tokens);
-        let container = events;
 
         switch (type) {
             case Event.Discover:
             case Event.Mutation:
-                // Layout events are queued based on the current configuration
-                // If lean mode is on, instead of sending these events to server, we back them up in memory.
-                // Later, if an upgrade call is called later in the session, we retrieve in memory backup and send them to server.
-                // At the moment, we limit backup to grow until MAX_BACKUP_BYTES. Anytime we grow past this size, we start dropping events.
+                // At the moment, we limit playback to grow until MAX_PLAYBACK_BYTES. Anytime we grow past this size, we start dropping events.
                 // This is not ideal, and more of a fail safe mechanism.
-                if (config.lean) {
-                    transmit = false;
-                    backupBytes += event.length;
-                    container = backupBytes < MAX_BACKUP_BYTES ? backup : null;
-                }
-                break;
-            case Event.Upgrade:
-                // As part of upgrading experience from lean mode into full mode, we lookup anything that is backed up in memory
-                // from previous layout events and get them ready to go out to server as part of next upload.
-                for (let entry of backup) { container.push(entry); }
-                backup = [];
-                backupBytes = 0;
+                playbackBytes += event.length;
+                if (playbackBytes < MAX_PLAYBACK_BYTES) { playback.push(event); }
```

Comment:
We should signal somehow that we have hit this limit and that the rest of the data is not complete.

Reply:
This is now new code addition and the comment explains it is not ideal. Something we can tackle in future PRs.